doc/source/ray-core/monitoring-debugging/tools.rst

```diff
@@ -0,0 +1,154 @@
+
+.. literalinclude:: ../doc_code/app_logging.py
+  :language: python
+
+All stdout emitted from the ``print`` method is printed to the driver with a ``(the task or actor repr, the process ID, IP address)`` prefix.
+
+.. code-block:: bash
+
+    (pid=45601) task
+    (Actor pid=480956) actor
+
+See :ref:`Logging <ray-logging>` for more details.
+
+Exceptions
+----------
+Creating a new task or submitting an actor task generates an object reference. When ``ray.get`` is called on the object reference,
+the API raises an exception if anything goes wrong with a related task, actor or object. For example,
+
+- :ref:`RayTaskError <ray-core-exceptions-ray-task-error>` is raised when there's an error from user code that throws an exception.
+- :ref:`RayActorError <ray-core-exceptions-ray-actor-error>` is raised when an actor is dead (by a system failure such as node failure or user-level failure such as an exception from ``__init__`` method). 
+- :ref:`RuntimeEnvSetupError <ray-core-exceptions-runtime-env-setup-error>` is raised when the actor or task couldn't be started because :ref:`a runtime environment <runtime-environments>` is failed to be created.
+
+See :ref:`Exceptions Reference <ray-core-exceptions>` for more details.
+
+Accessing Ray States
+--------------------
+Starting from Ray 2.0, it supports CLI / Python APIs to query the state of resources (e.g., actor, task, object, etc.).
+
+For example, the following command will summarize the task state of the cluster.
+
+.. code-block:: bash
+
+    ray summary tasks
+
+.. code-block:: text
+
+    ======== Tasks Summary: 2022-07-22 08:54:38.332537 ========
+    Stats:
+    ------------------------------------
+    total_actor_scheduled: 2
+    total_actor_tasks: 0
+    total_tasks: 2
+
+
+    Table (group by func_name):
+    ------------------------------------
+        FUNC_OR_CLASS_NAME        STATE_COUNTS    TYPE
+    0   task_running_300_seconds  RUNNING: 2      NORMAL_TASK
+    1   Actor.__init__            FINISHED: 2     ACTOR_CREATION_TASK
+
+The following command will list all the actors from the cluster.
+
+.. code-block:: bash
+
+    ray list actors
+
+.. code-block:: text
+
+    ======== List: 2022-07-23 21:29:39.323925 ========
+    Stats:
+    ------------------------------
+    Total: 2
+
+    Table:
+    ------------------------------
+        ACTOR_ID                          CLASS_NAME    NAME      PID  STATE
+    0  31405554844820381c2f0f8501000000  Actor                 96956  ALIVE
+    1  f36758a9f8871a9ca993b1d201000000  Actor                 96955  ALIVE
+
+See :ref:`Ray State API <state-api-overview-ref>` for more details.
+
+Dashboard (Web UI)
+------------------
+Ray supports the web-based dashboard to help users monitor the cluster. When a new cluster is started, the dashboard will be available
+through the default address `localhost:8265` (port can be incremented if port 8265 is already occupied).
```

Comment:
Present tense
the dashboard is available ...

Reply:
who increments the port? might want to say:
(if port 8265 is used, Ray increments to the next available port).